json_merger/dict_merger.py

```diff
@@ -154,16 +155,55 @@ def _merge_dicts(self):
         try:
             non_list_merger.run()
         except UnresolvedConflictsException as e:
-            non_list_merger.continue_run([self.pick
-                                          for i in range(len(e.content))])
-            for conflict in e.content:
+            continue_list = self._get_custom_strategies(
+                e.content
+            )
+            non_list_merger.continue_run(continue_list)
+            for i in range(len(e.content)):
+                conflict = e.content[i]
+                strategy = continue_list[i]
+
                 conflict_patch = {'f': conflict.second_patch,
-                                  's': conflict.first_patch}[self.pick]
+                                  's': conflict.first_patch}[strategy]
+
                 self.conflict_set.update(patch_to_conflict_set(conflict_patch))
 
         self._restore_lists()
         self.merged_root = patch(non_list_merger.unified_patches, self.root)
 
+    def _get_custom_strategies(self, patches):
+        strategies = []
+        for i in range(len(patches)):
+            head = patches[i].first_patch
+            strategies.append(
+                self._get_related_path(head[1])
+            )
+
+        return strategies
+
+    def _get_related_path(self, path):
+        def _get_path(custom_path):
+            if self.custom_ops.get(custom_path, None):
+                return {
+                    DictMergerOps.FALLBACK_KEEP_HEAD: 'f',
+                    DictMergerOps.FALLBACK_KEEP_UPDATE: 's'
+                }[self.custom_ops[custom_path]]
+            return None
+
+        if type(path) is list:
+            list_path = path[:-1]
+        else:
+            list_path = path.split('.')
+
+        for i in range(len(list_path)):
+            new_path = '.'.join(list_path)
+            path_found = _get_path(new_path)
+            if path_found:
+                return path_found
+            list_path.pop()
+
```

Comment:
I think this should then be factored out into a separate helper, because it's not obvious what it does.